crates/masonry/src/event_loop_runner.rs

```diff
@@ -5,123 +5,169 @@ use tracing::warn;
+                };
+                let delta = PhysicalPosition::new(delta.0, delta.1);
+                self.render_root
+                    .handle_pointer_event(PointerEvent::MouseWheel(
+                        delta,
+                        self.pointer_state.clone(),
+                    ));
+            }
+            _ => (),
+        }
+        while let Some(signal) = self.render_root.pop_signal() {
+            match signal {
+                render_root::RenderRootSignal::Action(action, widget_id) => {
+                    self.render_root.edit_root_widget(|root| {
+                        let mut driver_ctx = DriverCtx {
+                            main_root_widget: root,
                         };
-                        let delta = PhysicalPosition::new(delta.0, delta.1);
-                        render_root.handle_pointer_event(PointerEvent::MouseWheel(
-                            delta,
-                            main_state.pointer_state.clone(),
-                        ));
-                    }
-                    _ => (),
+                        self.app_driver
+                            .on_action(&mut driver_ctx, widget_id, action);
+                    });
+                }
+                render_root::RenderRootSignal::TextFieldAdded => {
+                    // TODO
+                }
+                render_root::RenderRootSignal::TextFieldRemoved => {
+                    // TODO
+                }
+                render_root::RenderRootSignal::TextFieldFocused => {
+                    // TODO
+                }
+                render_root::RenderRootSignal::ImeStarted => {
+                    // TODO
+                }
+                render_root::RenderRootSignal::ImeMoved => {
+                    // TODO
+                }
+                render_root::RenderRootSignal::ImeInvalidated => {
+                    // TODO
+                }
+                render_root::RenderRootSignal::RequestRedraw => {
+                    self.window.request_redraw();
+                }
+                render_root::RenderRootSignal::RequestAnimFrame => {
+                    // TODO
+                    self.window.request_redraw();
+                }
+                render_root::RenderRootSignal::SpawnWorker(_worker_fn) => {
+                    // TODO
+                }
+                render_root::RenderRootSignal::TakeFocus => {
+                    self.window.focus_window();
+                }
+                render_root::RenderRootSignal::SetCursor(cursor) => {
+                    self.window.set_cursor(cursor);
+                }
+                render_root::RenderRootSignal::SetSize(size) => {
+                    // TODO - Handle return value?
+                    let _ = self.window.request_inner_size(size);
+                }
+                render_root::RenderRootSignal::SetTitle(title) => {
+                    self.window.set_title(&title);
```

Comment:
These should probably use user events in winit, but that's for the future, not today.

Reply:
I think Masonry signals serve a different purpose.